app.py

```diff
@@ -15,9 +15,115 @@
+
+from thoth.common import init_logging
+from thoth.storages import GraphDatabase
+from thoth.storages import __version__ as __storages_version__
+from thoth.common import __version__ as __common_version__
+
+
+__version__ = "0.0.1"
+__service_version__ = f"{__version__}+" f"common.{__common_version__}.storages.{__storages_version__}"
+
+_LOGGER = logging.getLogger("thoth.graph_metrics_exporter")
+
+COMPONENT_NAME = "graph-metrics-exporter"
+
+# database
+KNOWLEDGE_GRAPH_HOST = os.getenv("KNOWLEDGE_GRAPH_HOST", "localhost")
+KNOWLEDGE_GRAPH_PORT = os.getenv("KNOWLEDGE_GRAPH_PORT", "5432")
+KNOWLEDGE_GRAPH_USER = os.getenv("KNOWLEDGE_GRAPH_USER", "postgres")
+KNOWLEDGE_GRAPH_DATABASE = os.getenv("KNOWLEDGE_GRAPH_DATABASE", "postgres")
+
+# metrics
+PROMETHEUS_REGISTRY = CollectorRegistry()
+THOTH_METRICS_PUSHGATEWAY_URL = os.environ["PROMETHEUS_PUSHGATEWAY_URL"]
+THOTH_DEPLOYMENT_NAME = os.environ["THOTH_DEPLOYMENT_NAME"]
+
+GRAPH_METRICS_EXPORTER_TASK = os.environ["THOTH_GRAPH_METRICS_EXPORTER_TASK"]
+
+REGISTERED_TASKS = ["graph_corruption_check"]
+
+init_logging()
+
+database_schema_revision_script = Gauge(
+    "thoth_database_schema_revision_script",
+    "Thoth database schema revision from script",
+    ["component", "revision", "env"],
+    registry=PROMETHEUS_REGISTRY,
+)
+
+graphdb_is_corrupted = Gauge(
+    "thoth_graphdb_is_corrupted",
+    "amcheck has detected corruption.",
+    [],
+    registry=PROMETHEUS_REGISTRY,
+)
+
+
+def _create_common_metrics(component_name: str = COMPONENT_NAME):
+    """Create common metrics to pushgateway."""
+    deployment_name = THOTH_DEPLOYMENT_NAME
+
+    if deployment_name:
+        database_schema_revision_script.labels(
+            component_name, GraphDatabase().get_script_alembic_version_head(), deployment_name
+        ).inc()
+    else:
+        _LOGGER.warning("THOTH_DEPLOYMENT_NAME env variable is not set.")
+
+
+def _send_metrics(component_name: str = COMPONENT_NAME):
+    """Send metrics to pushgateway."""
+    pushgateway_url = THOTH_METRICS_PUSHGATEWAY_URL
+    try:
+        _LOGGER.debug(f"Submitting metrics to Prometheus pushgateway {pushgateway_url}")
+        push_to_gateway(
+            pushgateway_url,
+            job=component_name,
+            registry=PROMETHEUS_REGISTRY,
+        )
+    except Exception as e:
+        _LOGGER.exception(f"An error occurred pushing the metrics: {str(e)}")
+
+
+def _graph_corruption_check(graph: GraphDatabase):
+    if graph.is_database_corrupted():
+        _LOGGER.info("Graph database is corrupted!")
+        graphdb_is_corrupted.set(1)
+    else:
+        _LOGGER.info("Graph database is not corrupted.")
+        graphdb_is_corrupted.set(0)
+
+
+def main():
+    """Perform graph metrics exporter job."""
+    _LOGGER.debug("Debug mode is on.")
+
+    _create_common_metrics()
+
+    graph = GraphDatabase()
+    graph.connect()
+
+    if GRAPH_METRICS_EXPORTER_TASK in REGISTERED_TASKS:
```

Comment:
You can use [click](https://click.palletsprojects.com/) as we do in other components to be consistent across the project. It will automatically do these type of checks for you if you annotate the main function.